packages/react-devtools-extensions/src/__tests__/updateMockSourceMaps.js

```diff
@@ -82,24 +96,35 @@ function compile(fileName) {
   );
   const encodedMappings = encode(decodedMappings);
 
-  // Generate compiled output with external inline base64 source maps
+  // Generate compiled output with external inline base64 source maps without column numbers
   writeFileSync(
-    resolve(inlineDir, fileName),
+    resolve(noColumnsDir, fileName),
     transformed.code +
       '\n//# sourceMappingURL=data:application/json;charset=utf-8;base64,' +
-      btoa(JSON.stringify(sourceMap)),
+      btoa(
+        JSON.stringify({
+          ...sourceMap,
+          mappings: encodedMappings,
+        }),
+      ),
     'utf8',
   );
 
-  // Generate compiled output with external inline base64 source maps without column numbers
+  // Generate compiled output with extended an extended sourcemap that
+  // includes a map of hook names.
+  const parsed = parse(code, {
+    sourceType: 'module',
+    plugins: ['jsx', 'flow'],
+  });
+  const encodedHookMap = generateEncodedHookMap(parsed);
   writeFileSync(
-    resolve(noColumnsDir, fileName),
+    resolve(extendedSourceMapDir, fileName),
     transformed.code +
       '\n//# sourceMappingURL=data:application/json;charset=utf-8;base64,' +
       btoa(
         JSON.stringify({
           ...sourceMap,
-          mappings: encodedMappings,
+          x_react_hook_map: [encodedHookMap],
```

Comment:
Naming nit: `x_react_hook_maps` (plural)

Reply:
I had been thinking we might use a more generic attribute like `x_react` as the top-level thing, inside of which we could put hook maps (for now) and maybe even expand later to include things like display name? I was thinking this may make it easier for tools like Metro to pass the info along, but maybe that's not true?

Reply:
1. Yes, adding a more generic mechanism for compilers/bundlers to represent source metadata is super desirable IMO. It would be a way to amortise the cost of teaching various tools to process this type of metadata (and in particular, keep it in sync with the rest of the data in the source map e.g. `sources` throughout the build process).
2. What you're proposing is virtually identical to `x_facebook_sources` under a different name. Support for this exists today (in Metro, in Hermes, in Facebook's internal infra, and some other tools like Sentry's [`rust-sourcemap`](https://github.com/getsentry/rust-sourcemap/search?q=x_facebook_sources)). We could, theoretically, publish an open spec for `x_facebook_sources` and incorporate this new React metadata there. From an implementor's standpoint, personally, I would prefer this option.
3. If we do want a separate React-branded field, `x_react_sources` would be my preferred naming - conveying that this is for source metadata and runs parallel to the `sources` array.

Reply:
Thinking a bit bigger than either (2) or (3), I can also see us designing an extensible metadata field that is more self-describing. (Names TBD)

```json5
{
  "x_sources_metadata_descriptors": ["metro:function-map", "react:hooks", /* ... other descriptors... */],
  "sources": ["file1.js", "file2.js"],
  "x_sources_metadata": [
    [
      /* Metro Function Map for file1.js */,
      /* React Hook Map for file1.js */,
      /* ... other metadata for file1.js ... */
    ],
    [
      /* Metro Function Map for file2.js */,
      /* React Hook Map for file2.js */,
      /* ... other metadata for file2.js ... */
    ]
  ]
}
```

Reply:
I like the more generic/extensible approach 👍🏼